src/main/java/com/nutomic/syncthingandroid/util/CustomX509TrustManager.java

```diff
@@ -0,0 +1,102 @@
+package com.nutomic.syncthingandroid.util;
+
+import android.util.Log;
+
+import org.apache.http.conn.ssl.SSLSocketFactory;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.Socket;
+import java.security.InvalidKeyException;
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.SignatureException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.X509TrustManager;
+
+
+/*
+ * TrustManager allowing the Syncthing https.pem CA
+ *
+ * Based on http://stackoverflow.com/questions/16719959/android-ssl-httpget-no-peer-certificate-error-or-connection-closed-by-peer-e/16759793#16759793
+ *
+ */
+public class CustomX509TrustManager implements X509TrustManager {
+
+    private static final String TAG = "CustomX509TrustManager";
+
+    /**
+     * Taken from: http://janis.peisenieks.lv/en/76/english-making-an-ssl-connection-via-android/
+     *
+     */
+    public static class CustomSSLSocketFactory extends SSLSocketFactory {
+        SSLContext sslContext = SSLContext.getInstance("TLS");
+
+        public CustomSSLSocketFactory(SSLContext context)
+                throws KeyManagementException, NoSuchAlgorithmException,
+                KeyStoreException, UnrecoverableKeyException {
+            super(null);
+            sslContext = context;
+        }
+
+        @Override
+        public Socket createSocket(Socket socket, String host, int port,
+                                   boolean autoClose) throws IOException {
+            return sslContext.getSocketFactory().createSocket(socket, host, port,
+                    autoClose);
+        }
+
+        @Override
+        public Socket createSocket() throws IOException {
+            return sslContext.getSocketFactory().createSocket();
+        }
+    }
+
+    private String mHttpsCertPath;
+
+    public CustomX509TrustManager(String httpsCertPath) {
+        mHttpsCertPath = httpsCertPath;
+    }
+
+    @Override
+    public void checkClientTrusted(X509Certificate[] chain, String authType)
+            throws CertificateException {
+    }
+
+    @Override
+    public void checkServerTrusted(java.security.cert.X509Certificate[] certs,
```

Comment:
I'm not comfortable with commenting code I did not wrote nor understand for 200%
(although it's functionality is clear for me as a user of this function)